tasks/connectors/edgescan/lib/edgescan_asset.rb

```diff
@@ -33,34 +34,34 @@ def location_specifiers
           end
         end
 
-        def find_location_specifier(specifier_id, location)
-          location_specifiers.find { |specifier| specifier.id == specifier_id } ||
-            location_specifiers.find { |specifier| specifier.location == location }
-        end
-
         # Converts an Edgescan asset into Kenna friendly ones
         #
-        # Edgescan and Kenna assets don't map one to one. A Kenna asset is more like an Edgescan
-        # location specifier. Because of that, one Edgescan asset usually gets turned into multiple
-        # Kenna assets.
-        #
         # This will:
-        # - Create Kenna assets based on Edgescan location specifiers
-        # - Go through the vulnerabilites and if some of them don't have a matching Edgescan
-        #   location specifier create Kenna assets for them
+        # - Create Kenna assets based on Edgescan hosts
+        # - If a matching host is not found an asset will be created from the vulnerability data
         def to_kenna_assets
-          location_specifiers_as_kenna_assets +
-            vulnerabilities_without_location_specifiers_as_kenna_assets
+          vulnerabilities.each_with_object([]) do |vulnerability, assets|
+            host = find_host(vulnerability.location)
+            asset = host ? host.to_corresponding_kenna_asset : vulnerability.to_corresponding_kenna_asset
+            assets << asset
+          end
         end
 
         private
 
-        def location_specifiers_as_kenna_assets
-          location_specifiers.map(&:to_kenna_asset)
+        def find_host(location)
```

Comment:
A Host's location will always be an IP. A Vulnerability's location could potentially be "https://test.example.com/some/page.html", for example.

Reply:
Makes sense - it might be a good idea to add nullable fields on the backend for things like URLs, hostnames, IP addresses, etc., but that's out-of-scope for this discussion - might make things easier for you and customers though.